scripts/mutate.py

```diff
@@ -253,18 +251,43 @@ def do_mutation():
         raise Exception(
             "Tried all possible combinations of modules and mutations and could not find a matching name"
         )
-    tests_failed = test_mutated_src()
+    tests_failed = test_mutated_src(args.test)
     yield (src, mutation_type, before, after, line), tests_failed
     restore(src)
 
 
+# Command line interface
+parser = argparse.ArgumentParser(description="Mutates Checker source code.")
+parser.add_argument(
+    "modules",
+    metavar="MODULE",
+    type=str,
+    nargs="+",
+    help="Module to mutate (eg. 'burrow.ml')",
+)
+
+parser.add_argument(
+    "--test",
+    metavar="COMMAND",
+    type=str,
+    help="Command to execute for testing the change",
+)
+
+parser.add_argument(
+    "--num-mutations", type=int, help="Number of mutations to test", default=25
+)
+
+
 if __name__ == "__main__":
-    n_mutations = 25
+    args = parser.parse_args()
+    random.seed(42)
```

Comment:
Random FYI - I don't think it matters in this case, but sometimes it can be useful to set this on the module level in case any random functions run on import or to avoid situations where you set the seed when calling main() but then forget to when importing a function from the module and calling it individually.

Reply:
Yeah, my idea here was to make it a command line argument actually. Likely when we put this into CI we might want to run it on multiple threads/times with different seed values. What would you do in that case? I guess we can always make it something like an environment variable, but that does not look as clean.

Reply:
You know, I hadn't really considered this in the context of randomized testing before :thinking:. I have mostly used seeds to work with random quantities for which you want to reproduce a single result (models, unit tests for some statistics quantity). I guess a question I have would be how much value running the script from multiple, fixed seeds has over increasing the sample size of a single run. Would this just be to improve performance (potentially more unique mutations for a fixed `n_mutations` or roughly the same number of mutations for less execution time) or is the goal to increase the sample size?

In the case of the latter, I would just crank up the `n_mutations` in a single process and leave the seed at the module level. I would also do the same if time isn't an issue and you just want to maximize coverage since running with different seeds doesn't necessarily mean that sample set you get with each different seed will be different. 

However, if performance in terms of CI run-time is the issue then I think you're approach of providing the seeds via the command line makes good sense since the mutations on each thread will probably at least have some differences (especially if the sample size is low relative to the total number of possible mutations).

Reply:
PS: Another option for running on multiple threads to get more samples for less wall time would be to pre-calculate the mutations with a single seed for a large `n_mutations` then multi-thread the actual application of the mutations (which would require creating copies of the repo for each thread).

Reply:
Interesting topic, this one. Personally I'd be fine either way, but I think that I prefer getting all the parameters from the command line. Then, the mutation testing tool looks more like a pure function, from a CI point of view.

> I guess a question I have would be how much value running the script from multiple, fixed seeds has over increasing the sample size of a single run. Would this just be to improve performance (potentially more unique mutations for a fixed `n_mutations` or more mutations for less wall time) or is the goal to increase the sample size?

This I am not sure about, but I have some feedback on the usability. Whcn checking `cfmm.ml`, to get past the "noise" (i.e., meaningless mutations which started accumulating at the top of the report) I ended up having to raise `n_mutations` up to 2000, which took a very long time to run. Locally I could also avoid this by keeping `n_mutations` low and changing the seed manually a few times, and to replicate this behavior on CI I'd expect the cleanest way to be to pass the seed via the command line.

Reply:
This branch was bit-rotting, so I merged it. I'm not sure about how to do this to be honest, I also think being able to provide a seed value would be useful, on the other hand @dorranh 's concern makes sense too. 

I guess we have to see whether we get the same results when we provide the same seed.